File: Basic Coding Problems/GCD or HCF/gcd.java
import java.util.*;

public class gcd {
    public static void main(String[] args) {
        int num1 = 12, num2 = 24, n = 1;
        
        if(num1!=num2){
            while (n != 0) {
                n = num1 % num2;
                if (n != 0) {
                    num1 = num2;
                    num2 = n;
                }
            }
            System.out.println("HCF : " + num2);
        }else{
            System.out.println("Invalid input");
        }


    }
}

File: Basic Coding Problems/GCD or HCF/gcd.py
num1,num2,n=10,15,1

if num1!=num2:
    while n!=0:
        n=num1%num2
        if n!=0:
            num1=num2
            num2=n
    print("HCF : ",num2)
else:
    print("Invalid input")